### ISSUE TITLE
I want a rule that will disallow patching `os.environ` with `mock.patch.dict`

### ISSUE THREAD

**Issue by harupy** [m-burst/flake8-pytest-style#252]
# Rule request

A rule to disallow using `unittest.mock.patch` to mock `os.environ`:

```python
from unittests import mock

def test_foo():
    with mock.patch.dict(os.environ, {"FOO": ...}):
        ...

    with mock.patch.dict(os.environ, {}):
        ...

```

Instead, we should use `monkeypatch.setenv` or `monkeypatch.delenv`:

```python
def test_foo(monkeypatch):
    monkeypatch.setenv("FOO", ...)
    ...

    monkeypatch.delenv("FOO")
    ...
```

## Description

```
```

## Rationale

```
```
